hack/check-code-patterns.sh

```diff
@@ -1,6 +1,20 @@
 #!/usr/bin/env bash
 
-set -eu
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eux
```

Comment:
set -x might be bit too much?
set -o pipefail is good to have here for sure.

Reply:
I'll remove the `-x`, but first I have to find out why this darn thing fails :imp: 
Yeah, and I forgot about `pipefail`

Reply:
So problem was that `grep` exits with 1, if it does not find a match. That does not play well with `set -eo pipefail`, so that I had to explicilty ignore a particular error code.

Reply:
`grep || true` is a fairly common pattern.